### ISSUE TITLE
#2 - MVP - Model View Presenter

### ISSUE THREAD

**Issue by benjaminfoo** [benjaminfoo/blog#2]
Hello world, 

today I wanted to share my thoughts about a basic architectural pattern called MVP, Model View Presenter.

**TL;DR - Overall:**
 - Applications who lack an seperation of layers or architectures are impossible to maintain and extend
 - MVP allows the separation of the presentation layer from the logic. Ideally the logic can be used for other presentation-layers without modifing it
 - Due to the fact that android couples presentation, interface and data access mechanisms we need to define well seperated layers
 - The implementation of the MVP pattern isn't and can't be defined - it depends on the amount of responsibilities that we code on the presenter
**

**TL;DR - The Presenter**
 - Acts as the middle man between view and model
 - Retrieves data from the model and returns it to the view
 - MVC unlike, it decides what happens when one interacts with the view

**TL;DR - The View**
 - Usually implemented by an activity, fragment, view.....
 - Contains a reference to the presenter (ideally provided by a DI framework like Dagger)
 - Only job: call a method of the presenter each time there is an interface action (click, touch, duh)

**TL;DR - The Model**
 - Provider of the data to display in the view
 - (In a very good layered architecture) Gateway for the domain layer or business logic

// TODO - Non tl;dr version

Source: [http://antonioleiva.com/mvp-android/](http://antonioleiva.com/mvp-android/)
